pkg/clusteragent/autoscaling/workload/controller.go

```diff
@@ -389,3 +388,35 @@ func (c *Controller) deletePodAutoscaler(ns, name string) error {
 	}
 	return nil
 }
+
+func (c *Controller) validateAutoscaler(podAutoscaler *datadoghq.DatadogPodAutoscaler) error {
+	// Check that targetRef is not set to the cluster agent
+	clusterAgentPodName, err := common.GetSelfPodName()
+	if err != nil {
+		return fmt.Errorf("Unable to get the cluster agent pod name: %w", err)
+	}
+
+	deploymentName := kubernetes.ParseDeploymentForPodName(clusterAgentPodName)
+	clusterAgentNs := common.GetMyNamespace()
+
+	if podAutoscaler.Namespace == clusterAgentNs && podAutoscaler.Spec.TargetRef.Name == deploymentName {
+		return fmt.Errorf("Autoscaling target cannot be set to the cluster agent")
+	}
+	return nil
+}
+
+func (c *Controller) updateAutoscalerStatus(ctx context.Context, key, ns, name string, err error, podAutoscalerInternal model.PodAutoscalerInternal, podAutoscaler *datadoghq.DatadogPodAutoscaler) error {
```

Comment:
I'd expect the method to highlight that it unlocks store and one line it:
```
		return autoscaling.NoRequeue, c.updateAndUnlock(ctx, key, ns, name, validationErr, podAutoscalerInternal, podAutoscaler)
```

There's probably a better name to find for it!